### ISSUE TITLE
[Audit] Are evaluation metrics computed before or after negative_sample_handling clipping?

### ISSUE THREAD

**Issue by grey-nearing** [google-research/flood-forecasting#296]
**Please confirm whether evaluation metrics are computed before or after `negative_sample_handling` clipping is applied, and document the answer.**

I don't know which it is — this is an audit request, not a bug report. But the two possibilities have quite different implications, so it seems worth pinning down explicitly rather than leaving implicit.

### Background

`negative_sample_handling` is opt-in. `Config.negative_sample_handling` defaults to `None`, and `_handle_negative_values` treats `''`/`'none'` as a no-op, so nothing clips unless a config asks for it.

Several shipped configs do ask for it:

```
pretrained-models/google-floodhub-settings-55-epochs/config.yml:83                  negative_sample_handling: clip
pretrained-models/...-nse-filtered-0.5-85-epochs/config.yml:83                      negative_sample_handling: clip
example-configs/floodhub-settings-config.yml:251                                    negative_sample_handling: clip
example-configs/handoff-forecast-lstm-config.yml:240                                negative_sample_handling: clip
example-configs/camels-multimet-mean-embedding-forecast-lstm-config.yml:155         negative_sample_handling: clip
```

When `clip` is active, `_handle_negative_values` applies `torch.clamp(values, min=normalized_zero)` inside the sampling path.

### The concern

Clipping is a monotone post-processing step that can only move predictions *towards* the observation whenever the observation is non-negative — which, for discharge, is always. So if metrics are computed on clipped predictions, then NSE, CRPS and friends are being improved by a post-processing step that is not part of the model.

That is entirely legitimate for **operational output**: you cannot publish a negative discharge forecast, and `max(Q(p), z)` is the honest way to enforce that.

It is more questionable for **evaluation**, for two reasons:

1. **Cross-model comparison.** If CMAL runs with `clip` and another head runs without it, the comparison is not like-for-like. The difference in reported skill would partly reflect a config flag rather than the models.
2. **It hides miscalibration.** Predictions below zero are a signal that the model is poorly calibrated at the bottom of the distribution. If that signal is clipped away before it reaches the metrics, the reported numbers look better precisely where the model is weakest, and the weakness stops being visible.

### What would resolve this

- Trace the path from `sample_pointpredictions` / `sample_cmal` / `sample_cmal_deterministic` through the evaluation code, and establish whether the tensor that reaches metric computation has had clipping applied.
- Document the answer in the `negative_sample_handling` docstring, so it is explicit either way.
- If metrics *are* computed post-clip, decide whether that is intended. Reporting both clipped and unclipped metrics would make the effect visible without giving anything up.

### Out of scope

This is only about where clipping sits relative to metric computation. Whether clipping should be the default, and whether the `cmal_deterministic` path honours the flag at all (#291), are separate questions.

<sub>Raised while reviewing #291. Analysis assisted by AI, reviewed by me.</sub>

**Comment by grey-nearing** [comment]
**Related note, separate from the audit question above: if we ever want the mixture mean clipped coherently, here is the closed form.**

Recording this here so it isn't lost. It came out of the #291 review but doesn't belong in that PR, since the gap it addresses is pre-existing and not that PR's to fix.

### The problem

`generate_predictions` returns 10 values: element 0 is the mixture **mean**, elements 1–9 are deciles.

Clipping the nine quantiles is coherent. Quantile functions are monotone, so `max(Q_X(p), z) == Q_max(X,z)(p)` — clipped quantiles really are the quantiles of the clipped variable. Verified numerically against 2e6 draws, agreement 0.0000 on every clipped quantile.

Element 0 is the exception. Clipping it yields `max(E[X], z)`, whereas the mean of the clipped variable is `E[max(X, z)]`. These differ by Jensen's inequality, and `max(E[X], z)` is not the mean of any distribution. Measured on three realistic mixtures with a `center=2.0, scale=4.0` scaler (`z = -0.5`):

| case | `max(E[X], z)` | `E[max(X, z)]` | gap |
|---|---|---|---|
| low flow | −0.338 | −0.053 | **−1.14 m³/s** |
| very low flow | −0.500 | −0.293 | **−0.83 m³/s** |
| mid flow | 0.515 | 0.753 | **−0.95 m³/s** |

So the mean element is reported on a different footing from the quantiles beside it.

### Closed form

The ALD mixture admits `E[max(X, z)]` analytically, in the repo's parameterization:

```python
# E[max(X, z)] for one ALD component
tau_c = 1 - tau
above = z + (b * tau_c / tau) * torch.exp(-tau * (z - mu) / b)
below = (
    mu
    - (tau * b / tau_c) * (1 - torch.exp((z - mu) * tau_c / b))
    + b * tau_c / tau
)
clipped_mean = (pi * torch.where(z > mu, above, below)).sum(-1)
```

Validated against 4e6 Monte Carlo draws: agreement ≤ 5e-4, i.e. within MC noise.

### Options, in my order of preference

1. Use the closed form above, so element 0 is the mean of the same clipped variable the quantiles describe.
2. Leave element 0 unclipped and document that it is the raw mixture mean.
3. Keep `max(E[X], z)`. This is the weakest option — it is neither the raw mean nor the clipped mean.

Not urgent, and not a regression: this behaviour predates #291. Filing it so the derivation is on record if anyone takes it up.

<sub>Analysis assisted by AI, reviewed by me.</sub>
